test/assert_diagnostic_SUITE.erl

```diff
@@ -0,0 +1,62 @@
+-module(assert_diagnostic_SUITE).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("stdlib/include/assert.hrl").
+-include_lib("src/assert_diagnostic.hrl").
+
+%% TestServer callbacks ++ test cases.
+-compile(export_all).
+
+groups() ->
+    [
+        {my_test_group, [parallel], [
+            test_equal,
+            test_expected_is_longer,
+            test_actual_is_longer,
+            test_one_distinc_item
+        ]}
+    ].
+
+all() ->
+    [{group, my_test_group}].
+
+
+%% Helper ensuring comparison of mismatching lists give the expected message.
+%% Implemented as a macro to have proper line report if the test case fails.
+-define(checkListEqualMessage(L0, L1, Msg),
```

Comment:
Could this be a function?

Reply:
I'd rather not!
- that's a very local helper, not meant to be used anywhere else anyway.
- with a function, the failure message wouldn't point to the test itself (would give live in function body rather line in test body).

Reply:
Okay fair "the failure message wouldn't point to the test itself" since we are only using it once, how about we inline it?
It would be really great if we can avoid macros as much as we can.

Reply:
Ok, turning that into a function!

Reply:
Note: my point about failure location is mostly irrelevant here, since the unit tests are very short and only test one thing each (as it should be!).